packages/syntax-core/src/useIsMounted.tsx

```diff
@@ -0,0 +1,13 @@
+import { useRef, useEffect } from "react";
+
+export default function useIsMounted(): React.MutableRefObject<boolean> {
+  const isMounted = useRef(false);
+
+  useEffect(() => {
+    isMounted.current = true;
+    return () => {
+      isMounted.current = false;
+    };
+  }, []);
+  return isMounted;
+}
```

Comment:
Naming nit, not blocking:

do you return `isMounted` ref directly because returning `isMounted.current` could have a stale value by the time it needs to be used?

If so, I think this hook should be named `useIsMountedRef`

```suggestion
export default function useIsMountedRef(): React.MutableRefObject<boolean> {
  const isMounted = useRef(false);

  useEffect(() => {
    isMounted.current = true;
    return () => {
      isMounted.current = false;
    };
  }, []);
  return isMounted;
}
```

Otherwise, for naming clarity, it feels like this hook should return `isMounted.current` directly.  What do you think?

```suggestion
export default function useIsMounted(): React.MutableRefObject<boolean> {
  const isMounted = useRef(false);

  useEffect(() => {
    isMounted.current = true;
    return () => {
      isMounted.current = false;
    };
  }, []);
  return isMounted.current;
}
```

Reply:
@benjroy @somethiiing changed this to an `isHydrated` hook which just returns a boolean. Feel free to take a look at other implementations:
* https://github.com/search?q=useIsHydrated&type=code 
* https://github.com/jmlweb/isMounted/blob/master/index.es.js
* https://gist.github.com/jaydenseric/a67cfb1b809b1b789daa17dfe6f83daa 
* https://github.com/webNeat/react-tidy/tree/master/src/useIsMounted 
* https://github.com/metric-space-ai/octopus_client/blob/c34583448cfc901e235bbf5837178abf25dd56b7/src/hooks/useIsHydrated.tsx#L10 

As for the previous questions, why not just return `isMounted.current` because in that case, it would be required to pass it as a dependency in any `useEffect`:
* https://www.codemzy.com/blog/ismounted-hook-with-useeffect-reactjs#adding-usecallback-to-create-a-custom-hook 
* https://github.com/jmlweb/isMounted/issues/1

Going to mark this as resolved since we no longer use a ref and just return a boolean